src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -42,6 +41,21 @@ public void intakeInit() {
         TalonFX.optimizeBusUtilizationForAll(intakeMotor);
         timer.stop();
         timer.reset();
+        intakeMotor.setInverted(true);
+    }
+
+    public void eject(double speed) {
+        intakeMotor.set(speed);
+        isEjecting = true;
+        switch (currentState) {
+            case HOLDING:
+                currentState = IntakeState.IDLE;
+        }
+    }
+
+    public void shoot(double speed) {
+        intakeMotor.set(speed);
+        currentState = IntakeState.EJECTING;
```

Comment:
Two things. One, if I'm calling an `eject` method and see that that's NOT the one that goes to the EJECT state but I see the `shoot` method does, I'm going to be very confused.

The second thing is that you're essentially trying to climb backwards in your state machine which isn't actually wrong per se, you just might have a harder time debugging things when your periodic may overwrite your state. ie, you move it back to IDLE and it moves it forward to INTAKING. Just something to think about, I know it works now.